.github/workflows/release-dataset.yml

```diff
@@ -0,0 +1,72 @@
+name: Publish dataset to production DB
+
+on:
+  workflow_dispatch:
+    branches: 
+      - main
+    inputs: 
+      megaset-name:
+        description: "Enter the name of the megaset to release every dataset within"
+        required: false
+      contained-dataset-id: 
+        description: "Enter the ID in the format [NAME]_v[VERSION] to release a single dataset contained within a megaset"
+        required: false
+      standalone-dataset-id: 
+        description: "Enter the ID in the format [NAME]_v[VERSION] to release a single standalone dataset"
+        required: false
+      
+
+jobs: 
+  release_dataset: 
+    name: Release Dataset
+    runs-on: ubuntu-latest
+    env:
+      NODE_ENV: "production"
+      FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+      FIREBASE_EMAIL: ${{ secrets.FIREBASE_EMAIL }}
+      AWS_SECRET: ${{ secrets.CFE_AWS_SECRET }}
+      AWS_ID: ${{ secrets.CFE_AWS_ID }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - run: npm ci
+      - name: Run release-dataset script
+        run: |
+          if: ${{ github.event.inputs.megaset-name }} && ${{ github.event.inputs.contained-dataset-id }}
+            npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.contained-dataset-id }}
+          elif [ -n "${{ github.event.inputs.megaset-name }}" ]; then
+            npm run release-dataset ${{ github.event.inputs.megaset-name }}
+          elif [ -n "${{ github.event.inputs.standalone-dataset-id }}" ]; then
+            npm run release-dataset ${{ github.event.inputs.standalone-dataset-id }}
+          else
+            echo "Error: either megaset name or dataset ID must be specified"
+            exit 1
+          fi
+      - name: Release every dataset in a megaset
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.contained-dataset-id == 'false' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }}
+      - name: Release a dataset contained within a megaset
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.contained-dataset-id == 'true' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.contained-dataset-id }}
+      - name: Release a dataset that is not part of a megaset
+        if: ${{ github.event.inputs.megaset-name == 'false' }} && ${{ github.event.inputs.standalone-dataset-id == 'true' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.standalone-dataset-id }}
+      - name: Error out if input variables are conflict   
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.standalone-dataset-id == 'true' }}
+        run: | 
```

Comment:
these won't be equal to `true`. are you trying to test if they exist?

Reply:
Yeah, I was trying to include the possible input combos and error out with clear messages if the combo has conflicts. I'll keep working on it and make these run statements cleaner.